pkgs/tools/filesystems/rar2fs/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv
+, fetchurl
+, fuse
+}:
+
+with stdenv.lib;
+
+let
+
+  version = "1.28.0";
+
+  unrar-src = fetchurl {
+    url = "https://www.rarlab.com/rar/unrarsrc-5.8.5.tar.gz";
+    sha256 = "0abvz6vv8kr416fphysfbwgxc6hyf1bpnd0aczfv7j3vc8x949d7";
+  };
+
+in
+
+stdenv.mkDerivation {
+  pname = "rar2fs";
+  inherit version;
+  inherit unrar-src;
+
+  prePhases = [ "prePhase" ];
+  prePhase = ''
+    tar xzvf ${unrar-src}
+    cd unrar
+    make lib
+  '';
+
+  buildInputs = [ fuse ];
+
+  configureFlags = [
+    "--with-unrar=/build/unrar/"
+    "--with-unrar-lib=/build/unrar/"
+  ];
+
+  src = fetchurl {
```

Comment:
`fetchFromGitHub` is better in this case, though both variants will do more or less the same thing ;)

Reply:
I'll fix!

Reply:
As soon as I introduce
```
  src = fetchFromGitHub {
    owner = "hasse69";
    repo = "rar2fs";
    rev = "b6d1d42952d8ec7153bc1c859de88b288cf6b5d6";
    sha256 = "0fmdqrs5yvn89ngc26vj5ggnalpwrdm8pdcfszw1wflh78hvd8kb";
  };
```
I get this error: `error: some outputs of '/nix/store/8zbcmjca0nr0r68jgpmay2z599hlx9fm-rar2fs-1.28.0.drv' are not valid, so checking is not possible` and I can't figure out why.

Reply:
When introducing 
```
  src = fetchFromGitHub {
    owner = "hasse69";
    repo = "rar2fs";
    rev = "b6d1d42952d8ec7153bc1c859de88b288cf6b5d6";
    sha256 = "0fmdqrs5yvn89ngc26vj5ggnalpwrdm8pdcfszw1wflh78hvd8kb";
  };
```
 it seems the working dir is `/build/source` instead of `/build/rar2fs-1.28.0` and it can't find the configure script
```
configuring
no configure script, doing nothing
```
as opposed to
```
configuring
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/wbwmdlmqys0gv311iylwifjq6wd5wiy0-rar2fs-1.28.0 --with-unrar=/build/unrar/ --with-unrar-lib=/build/unrar/
```

Any thoughts?

Reply:
Strange, this should normally work. For special-cases there is `sourceRoot` (https://nixos.org/nixpkgs/manual/#ssec-unpack-phase) but that shouldn't be required in this case (from a quick look at the GH repo).

Btw: You should use `rev = "v${version}` instead of the commit hash ;)

Also, it could be that `0fmdqrs5yvn89ngc26vj5ggnalpwrdm8pdcfszw1wflh78hvd8kb` is a wrong or old hash (you can use `nix-prefetch-url -A rar2fs.src` to get the correct one).

If it doesn't work feel free to push the current version, then I can have a look what's going on.

Reply:
Ok so I've figured out that tools like `automake`, `autoconf` and other utilities like `which` (which is needed by `get-version.sh`, no pun intended ;) ) are not readily available when using fetchFromGitHub instead of fetchurl so the configure phase does not succeed.

Is there any easy fix to this?

Reply:
Ok so I've found `autoreconfHook` which seems to do the magic. :)

Reply:
Sorry didn't see your last reply. I did figure the `rev = "v${version}"` out after some thinking :)
Thanks for the tip on with `nix-prefetch-url -A` instead of `nix-prefetch-url <URL>`